pkg/resource/sensitive.go

```diff
@@ -93,23 +87,35 @@ func GetConnectionDetails(attr map[string]interface{}, tr Terraformed, cfg *conf
 			// state sensitive keys and connection keys starting with this
 			// prefix are reserved and should not be used as a custom connection
 			// key.
-			return nil, errors.Errorf("custom connection keys cannot start with %q, overriding reserved connection key (%q) is not allowed", k, prefixAttribute)
+			return nil, errors.Errorf(errFmtCannotOverrideExistingKey, k)
+		}
+		if conn == nil {
+			conn = map[string][]byte{}
 		}
 		conn[k] = v
 	}
+
+	// NOTE(muvaf): Having ID in the connection details is not really a must but
+	// just a nice-to-have, hence we don't handle cases where the ID is given
+	// in a field other than "id". cfg.ExternalName.GetNameFn wouldn't really
+	// help because we need the raw ID.
+	if id, ok := attr["id"]; conn != nil && ok {
```

Comment:
Should we consider dropping including `id` to connection secrets? 

I remember we've added this to support cases where `id` field is required as a connection detail, e.g. id in [aws_iam_access_key](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_access_key#id). But now we have a way to include any information from the tf schema as a conn detail with `AdditionalConnectionDetailsFn`.

Reply:
Looks like we pass Terraform attribute map to `AdditionalConnectionDetailsFn`. `id` might incorporate addition info, such as the subscription id for Azure resources, which may not be available in any other attributes. We should consider this in the discussion.

Reply:
@ulucinar The attribute map we pass to `AdditionalConnectionDetailsFn` is the full TFState, which is where we pull the `id` from in this logic. So it should be available in that map. Maybe we can rename the argument name of that function from `attr` to `tfstate` to make it more clear.

@turkenh I thought about removing it but then ended up publishing it only if connection detail secret has some content so that we don't publish a connection secret for every resource, thinking that `id` of RDS Instance could be helpful for connecting for example but I do expect developers to write an `AdditionalConnectionDetailsFn` for _connectible_ resources in all cases anyway. I'll just remove it.